### PR TITLE
Expanded login below comments

### LINKED ISSUE #3756: Expanded login pane below comments
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://code.publiclab.org/#r=all), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @publiclab/reviewers to get it labelled. Or, if this is not your first time, try to solve it yourself!

****

When viewing a post while not logged in, you see a simple "log in to comment" message:

image

But we have a much friendlier log in pane in the navbar:

screenshot 2018-10-24 at 11 49 05 am

Let's use some of the same code from the navbar pane to make a friendlier area to log in (and especially quickly log in using Google/Facebook Oauth) below the comment area. 

The code for the navbar pane is here: 

https://github.com/publiclab/plots2/blob/b6e1647b929082bc54bbfa63fe4317d4ae3cf2c3/app/views/layouts/_header.html.erb#L124-L166

But in https://github.com/publiclab/plots2/issues/3755 it will be broken into a partial template, that could potentially be re-used. So this could be done after that issue!

### PR DESCRIPTION
**Pull request by JonathanXu1**
Fixes #3756 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [ ] ask `@publiclab/reviewers` for help, in a comment below

This pull request allows for a login pane below comments if the user is not logged in. Here is a screenshot:
image

### CODE REVIEW

**Comment by SidharthBansal** [comment]
Please rebase your work.

**Comment by jywarren** [comment]
Ah, would this also receive the same rearranging of icons and such as in
the other PR, since it's based on the same template? Thanks!

**Comment by jywarren** [comment]
And perhaps it should still say in small text -- "Please log in to
comment:" -- what do you think?

**Comment by JonathanXu1** [comment]
Yep, this would have the same styling as the other pull request. Rebasing now...

**Comment by SidharthBansal** [comment]
I agree with you @jywarren on 
>And perhaps it should still say in small text -- "Please log in to
comment:" -- what do you think?

**Comment by JonathanXu1** [comment]
Yep, there's a prototyps here: Merge conflict in app/views/user_sessions/_new.html.erb
How is it?

Also I'm getting the error after renaming new.html.erb to _new.html.erb. How to I fix the routing in rails?
image

**Comment by JonathanXu1** [comment]
image
@jywarren @SidharthBansal  how's this?

**Comment by jywarren** [comment]
I think it looks good, but perhaps it needs to have extra spaces around the OR - you could make it:

`> <span class="font-family:arial, sans-serif;"> OR </span> <`

to give it a little more room and make the font look like the standard text font instead of the title font?

**Comment by JonathanXu1** [comment]
image
@jywarren how's this?

**Comment by SidharthBansal** [comment]
Many changes which are not part of your pr are placed in this PR. Can you please remove them?

This looks perfect to me.

**Comment by JonathanXu1** [comment]
Sorry about that. I forgot to hard reset my master branch.

File: app/views/notes/_responses.html.erb [hunk 1]
```diff
@@ -32,7 +32,7 @@
         <%= render :partial => "comments/form", :locals => { title: 'Add feedback, suggestions, or photos
           ', body: 'I did this!', comment: false, placeholder: '' } %>
       <% else %>
-        <p><%= raw t('notes._comments.must_be_logged_in', :url1 => new_user_session_path( return_to: request.path )) %></p>
+        <%= render :partial => "user_sessions/new" %>
```
Comment (SidharthBansal) [inline app/views/notes/_responses.html.erb]: Is this part of this pr? OR work related to other PR?

File: app/views/notes/_comments.html.erb [hunk 2]
```diff
@@ -15,7 +15,7 @@
     <% if current_user %>
       <%= render :partial => "comments/form", :locals => { title: I18n.t('notes._comments.post_comment'), comment: false, placeholder: I18n.t('notes._comments.post_placeholder') } %>
     <% else %>
-    <p><%= raw t('notes._comments.must_be_logged_in', :url1 => new_user_session_path( return_to: request.path )) %></p>
+      <%= render :partial => "user_sessions/new" %>
```
Comment (SidharthBansal) [inline app/views/notes/_comments.html.erb]: Is it related to this PR OR other PR?

**Comment by SidharthBansal** [comment]
Please paste the final screenshot after you do the necessary changes

**Comment by SidharthBansal** [comment]
@JonathanXu1 can you please tell me the difference between #4133 and #4145. I think in both the PRs you are creating a modal. 
Gifs for both please so that we can review.

**Comment by JonathanXu1** [comment]
@SidharthBansal in comments and resopnses, I'm replacing the "you must be logged in to comment" on the bottom of the page with an expanded login below comments.
Before: 
image
After:
image

**Comment by JonathanXu1** [comment]
Here's the gif of the login pane at the bottom of the page: 
ezgif com-video-to-gif

The gif for the modal is on #4133 . There is no login modal as that will be included when #4133 is merged. Also the social icons are not displayed because the change to remove the development conditional is also made on the other PR.

**Comment by SidharthBansal** [comment]
Ok got it.
We need to merge #4133 first.

**Comment by SidharthBansal** [comment]
>the social icons are not displayed because the change to remove the development conditional is also made on the other PR.

It should now display. You need to rebase I guess.

**Comment by SidharthBansal** [comment]
@jywarren I may have a conflicting view. Sorry for that.
I don't think we need this pr now. We will be using the modal for this portion instead of expanded login.
@JonathanXu1 what is your suggestion?

**Comment by JonathanXu1** [comment]
Ok sure. Can I still submit this task on GCI since I've solved this issue with my PR?

**Comment by SidharthBansal** [comment]
If the task is no longer needed we will give you rewards and if we need this pr, then upon its merge we will give you credit.
But let's wait for the @jywarren to tell his views.
In the meantime you can finish up the sign-up modal and integrating both modals together.

**Comment by JonathanXu1** [comment]
Ah okay. What if I keep the original link on the bottom of the page and toggle the login modal when it's clicked?

**Comment by SidharthBansal** [comment]
Yeah, I also mean the same behaviour in above comments. But let's see
@jywarren viewpoint first.

If its about linking then also we will credit you for this PR. We will
credit you separately for linking PR if you do so after Jeff agrees to it.
Does it makes sense?

**Comment by JonathanXu1** [comment]
Sounds good :)

**Comment by SidharthBansal** [comment]
Please claim the task on the GCI dashboard, I will give you rewards. In case Jeff wants more changes then you can continue to work on this. I don't want that anyone will not be credited because their PR was not reviewed. That fault from our side. Sorry about that.

**Comment by JonathanXu1** [comment]
Will do. Though I don't think it's available on GCI anymore.

**Comment by SidharthBansal** [comment]
Please do it right now. I may not be available tomorrow. If @oorjitchowdhary @JonathanXu1 @dinaelhanan @okonek any of your task which you worked upon will not be approved today, I worry you will lose rewards regardless of the work you did.

**Comment by SidharthBansal** [comment]
@JonathanXu1 please check it. It should be available. If not please ping me. I will create the issue.

**Comment by JonathanXu1** [comment]
image
There's only the option of linking modals to comments. Don't think the original expanded login below comments task is available

**Comment by SidharthBansal** [comment]
https://codein.withgoogle.com/dashboard/tasks/5342380925386752/

**Comment by JonathanXu1** [comment]
Thanks!

**Comment by oorjitchowdhary** [comment]
> I don't think we need this pr now. We will be using the modal for this portion instead of expanded login.

Even I agree with that.. I actually just sent a patch for implementing the modal login on comments and questions

**Comment by dinaelhanan** [comment]
@SidharthBansal I have also been waiting for this to be approved. I don't know how to fix the errors, but I am running out of time on the task. I will continue to work on the errors even after the approval, it's just that today is last day:
https://codein.withgoogle.com/dashboard/task-instances/4824240835526656/

**Comment by SidharthBansal** [comment]
@jywarren must be busy somewhere, so I am waiting for his point. @Jeffrey
Warren <jywarren@gmail.com> you must be amazed by such huge number of tasks
all from me. :-) Sorry, to trouble you this much. :-))

**Comment by SidharthBansal** [comment]
Please don't make the conversations divergent.

**Comment by dinaelhanan** [comment]
@SidharthBansal I didn't mean to make it divergent, you tagged me asking if I needed here:
> Please do it right now. I may not be available tomorrow. If @oorjitchowdhary @JonathanXu1 @dinaelhanan @okonek any of your task which you worked upon will not be approved today, I worry you will lose rewards regardless of the work you did.
:)

**Comment by jywarren** [comment]
Hello all - yes, this happened SO FAST and there've been SO MANY things getting done that I couldn't keep up! Please forgive me! And thank everyone for your patience and understanding as so many issues/PRs/tasks got made, solved, approved, reviewed, etc in the past few days. I'm sure it's been trying!

**Pull request closed without merging**